### ISSUE TITLE
Invalid setter generated for member reference to type with deleted operator=

### ISSUE THREAD

**Issue by dbankieris** [swig/swig#2866]
```c++
%module test

%inline %{

class Foo {
    Foo& operator=(const Foo&) = delete;
};

class Bar {
    public:
    Foo foo;
    Foo& fooRef{foo};
};

%}
```
```bash
$ swig -c++ -python test.i
$ g++ `python3-config --includes` -fPIC -shared -o test.so test_wrap.cxx
test_wrap.cxx: In function ‘PyObject* _wrap_Bar_fooRef_set(PyObject*, PyObject*)’:
test_wrap.cxx:2876:31: error: use of deleted function ‘Foo& Foo::operator=(const Foo&)’
 2876 |   if (arg1) (arg1)->fooRef = *arg2;
      |                               ^~~~
test_wrap.cxx:2770:10: note: declared here
 2770 |     Foo& operator=(const Foo&) = delete;
      |          ^~~~~~~~

$ swig -version
SWIG Version 4.0.1
Compiled with g++ [x86_64-pc-linux-gnu]
Configured options: +pcre
```

Note that a setter is not generated for the actual instance `foo`, only the reference `fooRef`.

**Comment by ojwb** [comment]
4.0.1 is approaching 5 years old, but I checked and this is still reproducible with git master.  It's not specific to Python (same issue with `-php` at least).

**Comment by wsfulton** [comment]
You can work around this by manually suppressing the setter with:
```swig
%immutable Bar::fooRef;
```
I've marked for fixing it in the next version of SWIG where the setter will be automatically suppressed.

**Comment by ojwb** [comment]
> I've marked for fixing it in the next version of SWIG where the setter will be automatically suppressed.

@wsfulton Are you saying you pushed a fix, or just planned to fix this for 4.3.0?  I still seem to be able to reproduce with 5a1d7ae0355c9985d3af9bc17bc791bf0208f2c6.

**Comment by wsfulton** [comment]
No fix available yet. At the time I figured it would be easy enough to do for 4.3, hence I added it to the 4.3 milestone, so I'll still try for 4.3, but no promises!
